app/src/main/kotlin/com/saltedge/authenticator/features/connections/actions/ActionFragment.kt

```diff
@@ -0,0 +1,147 @@
+import com.saltedge.authenticator.interfaces.UpActionImageListener
+import com.saltedge.authenticator.sdk.model.ActionDeepLinkData
+import com.saltedge.authenticator.tool.*
+import com.saltedge.authenticator.widget.fragment.BaseFragment
+import kotlinx.android.synthetic.main.fragment_action.*
+import kotlinx.android.synthetic.main.fragment_connect_processing.view.*
+import javax.inject.Inject
+
+const val KEY_ACTION_DEEP_LINK_DATA = "ACTION_DEEP_LINK_DATA"
+
+class ActionFragment : BaseFragment(),
+    ActionContract.View,
+    UpActionImageListener,
+    View.OnClickListener {
+
+    @Inject
+    lateinit var presenterContract: ActionContract.Presenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        injectDependencies()
+        arguments?.getString(KEY_GUID)?.let {
+            presenterContract.setInitialData(
+                connectionGuid = it,
+                actionDeepLinkData = arguments?.getSerializable(KEY_ACTION_DEEP_LINK_DATA) as ActionDeepLinkData
+            )
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        activityComponents?.updateAppbarTitleWithFabAction(getString(presenterContract.getTitleResId()))
+        return inflater.inflate(R.layout.fragment_action, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        completeView?.setOnClickListener(this)
+        updateViewsContent()
+        presenterContract.viewContract = this
+        presenterContract.onViewCreated()
+    }
+
+    override fun onDestroyView() {
+        presenterContract.viewContract = null
+        super.onDestroyView()
+    }
+
+    override fun onDestroy() {
+        presenterContract.onDestroyView()
+        super.onDestroy()
+    }
+
+    override fun updateViewsContent() {
+        completeView?.setIconResource(presenterContract.iconResId)
+        completeView?.setTitleText(presenterContract.completeTitle)
+        completeView?.setSubtitleText(presenterContract.completeMessage)
+        completeView?.setMainActionText(presenterContract.mainActionTextResId)
+        fragmentActionProcessing.titleView.text = getString(R.string.action_status_in_progress)
+
+        updateLayoutsVisibility()
+    }
+
+    override fun getUpActionImageResId(): ResId? = R.drawable.ic_close_white_24dp
+
+    override fun closeView() {
+        activity?.finishFragment()
+    }
+
+    override fun showErrorAndFinish(message: String) {
+        activity?.showWarningDialog(
+            message = message,
+            listener = DialogInterface.OnClickListener { _: DialogInterface, _: Int ->
+                activity?.finishFragment()
+            }
+        )
+    }
+
+    override fun returnActionWithConnectionId(authorizationID: String, connectionID: String) {
```

Comment:
let extract connectionID: ConnectionID and authorizationID: AuthorizationID to `authorizationIdentifier`